### ISSUE TITLE
A second instance of a notebook is opened when debugging with two editor groups

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#133761]
- Open a notebook in one editor group 1, and a editor group 2 with anything in it
- Start RBL in the notebook
- Focus group 2
- Open the debug view, click the stack frame for the notebook cell (there are other ways to make this happen too)
- A second notebook editor instance is opened in group 2

We just call openEditor and it should reveal the cell in the editor where it's already open. But I think that we can't do this correctly for notebook cells. Where is the code that checks whether the editor is already open?

![image](https://user-images.githubusercontent.com/323878/134716924-05e817a8-b18c-4903-b3f1-303bd849f6c7.png)

**Comment by lramos15** [comment]
Simpler repo without debugging.

1. Open a github issue notebook
2. Create an error
3. Focus a different group
4. Click the problem in the problems panel
5. 🐛 New notebook appears

**Comment by bpasero** [comment]
This is probably a duplicate of https://github.com/microsoft/vscode/issues/131619

Here is what happens:
* you click on a problem
* we open an untyped input as shown below
* we ask the notebook if it matches that input 
* notebook says, it doesn't match
* we consider the right hand empty group as target

![image](https://user-images.githubusercontent.com/900690/134724045-a9f706fa-6389-4184-8124-0644b8389de9.png)

**Comment by lramos15** [comment]
> * we ask the notebook if it matches that input

This sounds like an easy enough fix then. The notebook knows how to translate a cell URI into the notebook URI. That's what it should use to determine the group

**Comment by bpasero** [comment]
Yeah it is a workaround if it is possible for a typed notebook input to know the cell it contains, would have to go somewhere here:

https://github.com/microsoft/vscode/blob/aaa8e4dfc39e1de828ed17853b01498b460be4db/src/vs/workbench/contrib/notebook/common/notebookEditorInput.ts#L285-L287

**Comment by roblourens** [comment]
So `matches` should return true if the other is a cell inside this notebook? That's easy enough but could there be any other consequences that I should watch out for?

**Comment by lramos15** [comment]
I don't think there's any consequences as you effectively want each cell to be considered the same input as the parent notebook.

**Comment by lramos15** [comment]
Reopening because fix was reverted

**Comment by lramos15** [comment]
@bpasero Talked with @rebornix a bit and matches cannot match for cell uris in this case as they actually want to create and resolve a new editor input which we do not do if the active editor matches the passed in one as it is normally unnecessary work. I was wondering if they should add special logic here https://github.com/microsoft/vscode/blob/34a321e2cab8d0f1992bcbca8d05ea7153e00125/src/vs/workbench/services/editor/common/editorGroupFinder.ts#L196-L211 to have the notebook and cell uri match so that the group chosen is correct. I know it's not ideal because it is notebook specific

**Comment by bpasero** [comment]
Would https://github.com/microsoft/vscode/issues/131619 address this too?

**Comment by lramos15** [comment]
@bpasero I think so as the cell uri is finding the incorrect target group. If the editor input created was opened in the correct group it would simply reveal the notebook and scroll it as expected.

**Comment by roblourens** [comment]
I don't understand what has to be done here. Any change a year later?

**Comment by lramos15** [comment]
@roblourens I don't believe we will be able to fix this until https://github.com/microsoft/vscode/issues/131619 is addressed

**Comment by roblourens** [comment]
Why is that one related? 

For future reference, here are my notes from our call a couple weeks ago

- In NotebookServiceImpl, add a separate editor registration for the Untitled- pattern, and look up the matching open editor input, set the cellOptions to reveal the cell, and return that
- Could possibly reduce the number of factories by just registering one for all registered editor types, in one glob

**Comment by lramos15** [comment]
> Why is that one related?
> 
> For future reference, here are my notes from our call a couple weeks ago
> 
> * In NotebookServiceImpl, add a separate editor registration for the Untitled- pattern, and look up the matching open editor input, set the cellOptions to reveal the cell, and return that
> * Could possibly reduce the number of factories by just registering one for all registered editor types, in one glob

Those notes are for the untitled issue this is for something else. The issue here is that the Cell URI doesn't match the notebook URI in the matches call so we improperly select the second group and then it resolves to the exact same notebook causing two to be opened. The reason we have to call this matches before resolution is complete and not after is due to webviews needing to know what group they belong to. Ideally an editor input is group agnostic which is why we hope @mjbvz can eventually resolve #131619 as there are some issues that just cannot be fixed in a proper manner without it,

**Comment by roblourens** [comment]
Oops, thanks. https://github.com/microsoft/vscode/issues/166161
